src/javascript/app/Modules/SmartChart/Containers/smart-chart.jsx

```diff
@@ -38,6 +38,15 @@ class Chart extends React.Component {
     );
 
     render() {
+        const granularity = this.props.granularity;
+        const duration    = this.props.end_epoch - this.props.start_epoch;
+
+        const margin = granularity === 0 ? (Math.max(300, (30 * duration) / (60 * 60) || 0)) : 3 * granularity;
```

Comment:
@negarn when the granularity is `undefined`, the value of margin become `NaN`.  Also, you should pass margins when you're passing startEpoch or endEpoch in other cases you shouldn't pass them to the chart.

Reply:
Maybe we can add another setter in `smartcharts` store and pass the `prop` when the contract is mounted.